src/widgets/unfold_widgets/import_widget.py

```diff
@@ -130,21 +141,29 @@ def clear_widgets(self):
         for i in reversed(range(self.columns_group_form_layout.count())):
             self.columns_group_form_layout.itemAt(i).widget().setParent(None)
 
-    def set_columns_grid(self):
+    def set_columns_grid(self, columns=None):
         """ draw columns and checkbox to choose them """
-        columns = self.engine.get_columns()
+        if columns is None:
+            columns = self.engine.get_columns()
 
         for column in columns:
             checkbox = QCheckBox(column)
             checkbox.setMinimumHeight(26)
             checkbox.setChecked(True)
             self.columns_group_form_layout.addRow(checkbox)
 
+        self.columns_button.setEnabled(True)
+
     def display_data(self):
         self.engine.read_data(self.get_checked_columns())
         if (data := self.engine.imported_data) is not None:
             self.data_table.setModel(QtTable(data))
 
+    def update_data_view(self):
+        if (data := self.engine.state.imported_data) is not None:
+            self.data_table.setModel(QtTable(data))
+        self.set_columns_grid(self.engine.state.imported_data.columns)
```

Comment:
self.clear_widgets probably should be used here too, because old columns would remain if we imported some set earlier.

Reply:
`self.set_columns_grid` already removes old columns before setting new ones, so I don't think it's necessary in this particular case